### ISSUE TITLE
 Incompatible character encodings: ASCII-8BIT and UTF-8 , after updating model.

### ISSUE THREAD

**Issue by Zhigun** [rails/rails#27210]
### Steps to reproduce

Create a model with tags column like in example above and save it. Then update the column tags with a word that contains 'russian' letters only(or partially) and save it. The result will be our object with tags in ASCII-8BIT encoding.
```
begin
  require "bundler/inline"
rescue LoadError => e
  $stderr.puts "Bundler version 1.10 or later is required. Please update your Bundler"
  raise e
end

gemfile(true) do
  source "https://rubygems.org"
  gem 'rails', '5.0.0.beta2'
  gem "pg"
end

require "active_record"
require "minitest/autorun"
require "logger"

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: "postgresql", database: ":memory:")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :instructors, force: true do |t|
    t.string :tags, array: true, default: []
  end
end

class Instructor < ActiveRecord::Base
end

class BugTest < Minitest::Test
  def test_association_stuff
    instructor = Instructor.create!(tags: ['asas', 'asdada'])
    assert_equal ['asas', 'asdada'], instructor.tags

    instructor.tags << 'Привет'
    instructor.save
    same_instructor_loaded_agin = Instructor.find(instructor.id)
    assert_equal ['asas', 'asdada', 'Привет'], same_instructor_loaded_agin.tags
    assert_equal ['asas', 'asdada', 'Привет'], instructor.tags
  end
end
```
### Expected behavior
It should be encoded correctly.
For example if we update with ['привет', 'tag1', 'tag2'] the same array is expected to be mapped back for model.

### System configuration
rails 5.0.0.beta2
ruby ruby 2.3.0p0

**Comment by sgrif** [comment]
This was already fixed by https://github.com/rails/rails/commit/a710dbd9, which will be included in 5.0.1

**Comment by Zhigun** [comment]
Thank you!
